File: experiments_records/1_latent vulnerabilities/2_requirement_level/python/educollab_flask_req05_repair_pass/models/assignment_repo.py
from models.db import get_db


def create_assignment(course_id, created_by, title, description, due_at):
    db = get_db()
    cursor = db.execute(
        'INSERT INTO assignments (course_id, created_by, title, description, due_at) VALUES (?, ?, ?, ?, ?)',
        (course_id, created_by, title, description, due_at),
    )
    db.commit()
    return get_assignment_by_id(course_id, cursor.lastrowid)


def list_assignments(course_id):
    db = get_db()
    return db.execute(
        'SELECT a.assignment_id, a.course_id, a.created_by, a.title, a.description, a.due_at, a.created_at, a.updated_at, '
        'u.username, u.display_name '
        'FROM assignments a JOIN users u ON u.user_id = a.created_by '
        'WHERE a.course_id = ? ORDER BY a.assignment_id DESC',
        (course_id,),
    ).fetchall()


def get_assignment_by_id(course_id, assignment_id):
    db = get_db()
    return db.execute(
        'SELECT a.assignment_id, a.course_id, a.created_by, a.title, a.description, a.due_at, a.created_at, a.updated_at, '
        'u.username, u.display_name '
        'FROM assignments a JOIN users u ON u.user_id = a.created_by '
        'WHERE a.course_id = ? AND a.assignment_id = ?',
        (course_id, assignment_id),
    ).fetchone()


def update_assignment(course_id, assignment_id, title, description, due_at):
    db = get_db()
    db.execute(
        'UPDATE assignments SET title = ?, description = ?, due_at = ?, updated_at = CURRENT_TIMESTAMP WHERE course_id = ? AND assignment_id = ?',
        (title, description, due_at, course_id, assignment_id),
    )
    db.commit()
    return get_assignment_by_id(course_id, assignment_id)


def delete_assignment(course_id, assignment_id):
    db = get_db()
    db.execute('DELETE FROM assignments WHERE course_id = ? AND assignment_id = ?', (course_id, assignment_id))
    db.commit()

File: EduCollab_benchmark/4_function_level/EduColla-js/educollab_JS/src/repos/assignmentRepo.js
const { db, nowIso } = require('../db');

function createAssignment({ course_id, created_by, title, description, due_at }) {
  const ts = nowIso();
  const info = db.prepare(
    'INSERT INTO assignments (course_id, created_by, title, description, due_at, created_at, updated_at) VALUES (?,?,?,?,?,?,?)'
  ).run(course_id, created_by, title, description, due_at, ts, ts);
  return getById(course_id, info.lastInsertRowid);
}

function listAssignments(course_id) {
  return db.prepare(`
    SELECT a.*, u.username as created_by_username
    FROM assignments a
    JOIN users u ON u.user_id = a.created_by
    WHERE a.course_id=?
    ORDER BY a.assignment_id DESC
  `).all(course_id);
}

function getById(course_id, assignment_id) {
  return db.prepare(`
    SELECT a.*, u.username as created_by_username
    FROM assignments a
    JOIN users u ON u.user_id = a.created_by
    WHERE a.course_id=? AND a.assignment_id=?
  `).get(course_id, assignment_id);
}

function updateAssignment(course_id, assignment_id, { title, description, due_at }) {
  db.prepare('UPDATE assignments SET title=?, description=?, due_at=?, updated_at=? WHERE course_id=? AND assignment_id=?')
    .run(title, description, due_at, nowIso(), course_id, assignment_id);
  return getById(course_id, assignment_id);
}

function deleteAssignment(course_id, assignment_id) {
  db.prepare('DELETE FROM assignments WHERE course_id=? AND assignment_id=?').run(course_id, assignment_id);
}

module.exports = { createAssignment, listAssignments, getById, updateAssignment, deleteAssignment };
